### ISSUE TITLE
linter: vitest/valid-expect false-positive shouldn't warn nor fix lately awaited one

### ISSUE THREAD

**Issue by Hideman42** [oxc-project/oxc#23815]
### What version of Oxlint are you using?

latest

### What command did you run?

oxlint --fix --fix-suggestions

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["vitest"],
  "rules": {
    "vitest/valid-expect": "error"
  }
}
```

### What happened?

The current code is valid the promise is awaited after the trigger and has to be started early to record the trigger.
This case is false-positive and rule should detect if this is not awaited at all.

<img width="1138" height="228" alt="Image" src="https://github.com/user-attachments/assets/8c6714ee-8a5d-4620-84e0-6fa2e38bf077" />

**Comment by durationextender** [comment]
@camc314 can i work on this issue?

**Comment by camc314** [comment]
Hi, i need to double-check.

If memory serves, i believe we are already aligned with eslint plugin vitest here.

**Comment by durationextender** [comment]
alright please let me know !
